src/rust-crypto/backup.ts

```diff
@@ -280,6 +308,93 @@ export class RustBackupManager extends TypedEventEmitter<RustBackupCryptoEvents,
             }
         }
     }
+
+    /**
+     * Creates a new key backup by generating a new random private key.
+     * If there is an existing backup server side it will be deleted and replaced
+     * by the new one.
+     *
+     * @param signer - Method that should sign the backup with existing device and
+     * existing identity.
+     * @returns a KeyBackupCreationInfo - All information related to the backup.
+     */
+    public async setUpKeyBackup(signer: (authData: AuthData) => Promise<void>): Promise<KeyBackupCreationInfo> {
+        // check first if there is an existing backup
+        if (this.activeBackupVersion || (await this.requestKeyBackupVersion()) != null) {
+            // we want to delete?
+            await this.deleteKeyBackup();
+        }
+
+        const version = await this.prepareKeyBackupVersion();
+        await signer(version.auth_data);
+
+        const res = await this.http.authedRequest<{ version: string }>(
+            Method.Post,
+            "/room_keys/version",
+            undefined,
+            {
+                algorithm: version.algorithm,
+                auth_data: version.auth_data,
+            },
+            {
+                prefix: ClientPrefix.V3,
+            },
+        );
+
+        this.olmMachine.saveBackupDecryptionKey(version.decryptionKey, res.version);
+
+        return {
+            version: res.version,
+            algorithm: version.algorithm,
+            authData: version.auth_data,
+            decryptionKey: version.decryptionKey,
+        };
+    }
+
+    /**
+     * Deletes all key backups.
+     * Will call the API to delete active backup until there is no more present.
+     */
+    public async deleteKeyBackup(): Promise<void> {
+        // there could be several backup versions, delete all to be safe.
+        let current = (await this.requestKeyBackupVersion())?.version ?? null;
+        while (current != null) {
+            await this.deleteKeyBackupVersion(current);
+            current = (await this.requestKeyBackupVersion())?.version ?? null;
+        }
+
+        // Should this also update Secret Storage and delete any existing keys?
+    }
+
+    /**
+     * Deletes the given key backup.
+     *
+     * @param version - The backup version to delete.
+     */
+    public async deleteKeyBackupVersion(version: string): Promise<void> {
+        logger.debug(`deleteKeyBackupVersion v:${version}`);
+        const path = encodeUri("/room_keys/version/$version", { $version: version });
+        await this.http.authedRequest<void>(Method.Delete, path, undefined, undefined, {
+            prefix: ClientPrefix.V3,
+        });
+    }
+
+    /**
+     * Prepare the keybackup version data, auth_data not signed at this point
+     * @returns a {@link PreparedKeyBackupVersion} with all information about the creation.
+     */
+    private async prepareKeyBackupVersion(): Promise<PreparedKeyBackupVersion> {
```

Comment:
given that this doesn't actually create a `version`, I'm not really convinced it is preparing a key backup version. It feels like it needs renaming and a clearer description:

```suggestion
    /**
     * Generate a new random key for backup.
     *
     * @returns a {@link GeneratedBackupKey} with all information about the creation.
     */
    private async generateBackupKey(): Promise<GeneratedBackupKey> {
```

Reply:
It's doing more than generating a backup key, it returns what you need to create the backup (algorithm, auth_data). Prepare is not that bad in that sense (and same naming that legacy code)